lib/Alcaeus/MongoDbAdapter/AbstractCursor.php

```diff
@@ -127,15 +133,22 @@ public function key()
     }
 
     /**
-     * Advances the cursor to the next result
+     * Advances the cursor to the next result, and returns that result
      * @link http://www.php.net/manual/en/mongocursor.next.php
      * @throws \MongoConnectionException
      * @throws \MongoCursorTimeoutException
-     * @return void
+     * @return array Returns the next object
```

Comment:
MongoCursor::next specifies the return type to be array:
http://www.php.net/manual/en/mongocursor.next.php

And that is also how it is implemented:
https://github.com/mongodb/mongo-php-driver-legacy/blob/master/cursor.c#L981

I was very confused by this as well but MongoCursor is not implemented like a standard Iterator.